### PR TITLE
WIP: Async db refactor

### PR DESCRIPTION
**Pull request by lfdversluis**
First async DB rewriting! 

Added some tests for the stuff I rewrote + one additional in the file that was not yet covered.

### CODE REVIEW

**Comment by lfdversluis** [comment]
@whirm Do you suggest that all DB async rewriting go into this PR or create several PRs with changes? Also taking the fact into account that this will need to be (somehow) tested on performance gain percentage.

File: Tribler/Test/Core/test_misc_utilities.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        self.config.set_videoplayer(False)
+        self.session = Session(self.config, ignore_singleton=True)
+        self.session.sqlite_db = SQLiteCacheDB(self.session)
+        self.db_path = u":memory:"
+        self.session.sqlite_db.initialize(self.db_path)
+
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+        d = defer.maybeDeferred(printDBStats, self._logger, self.session)
+        d.addErrback(log.err)
```
Comment (whirm) [inline Tribler/Test/Core/test_misc_utilities.py]: This is swallowing the error, you want the test to fail if there's one

File: Tribler/Test/Core/test_misc_utilities.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        self.config.set_torrent_collecting(False)
+        self.config.set_libtorrent(False)
+        self.config.set_dht_torrent_collecting(False)
+        self.config.set_videoplayer(False)
+        self.session = Session(self.config, ignore_singleton=True)
+        self.session.sqlite_db = SQLiteCacheDB(self.session)
+        self.db_path = u":memory:"
+        self.session.sqlite_db.initialize(self.db_path)
+
+        self._logger = logging.getLogger(self.__class__.__name__)
```
Comment (whirm) [inline Tribler/Test/Core/test_misc_utilities.py]: The setUp method is doing this for you already

**Comment by lfdversluis** [comment]
retest this please

**Comment by synctext** [comment]
retest this please

**Comment by lfdversluis** [comment]
Out of date, StormDBManager will do this in the near future.

**Pull request closed without merging**